x-pack/platform/plugins/shared/alerting/server/lib/rule_gaps/update/utils.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { BackfillSchedule } from '../../../application/backfill/result/types';
+import { parseDuration } from '../../../../common';
+import { clipDateIntervals } from '../gap/interval_utils';
+import type { Gap } from '../gap';
+
+export interface ScheduledItem {
+  from: Date;
+  to: Date;
+  status: BackfillSchedule['status'];
+}
+
+export const toScheduledItem = (backfillSchedule: BackfillSchedule): ScheduledItem => {
+  const runAt = new Date(backfillSchedule.runAt).getTime();
+  const intervalDuration = parseDuration(backfillSchedule.interval);
```

Comment:
`parseDuration` can throw an error, should there be an attempt to catch and handle?